users/models.py

```diff
@@ -274,3 +275,41 @@ def create_last_seen_for_user(sender, instance, created, raw, **kwargs):
 
     if created:
         LastSeen.objects.create(user=instance, at=timezone.now())
+
+
+class InvitationToken(models.Model):
+    """
+    An invitation token to create a user account.
+    """
+
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        related_name="invitations",
+        help_text="The user who was invited to create an account.",
+    )
+    token = models.UUIDField(
+        unique=True,
+        default=uuid.uuid4,
+        help_text="A unique, random UUID that represents the invitation token.",
+    )
+    expiration_date = models.DateTimeField(
+        help_text="The date and time when the token expires."
+    )
+
+    def __str__(self):
+        return f"{self.user.get_display_name}'s Invitation Token"
+
+    @property
+    def is_expired(self):
+        """Returns True if the token is expired, False otherwise."""
+        return datetime.datetime.now() > self.expiration_date
```

Comment:
Since this file is already using `now()` from django utils' timezone, do you think we could use that here as well?

Also django's `now` is timezone aware while datetime's is not (https://stackoverflow.com/questions/26949959/timezone-now-vs-datetime-datetime-now), and timezone's `now` will DTRT depending on the `USE_TZ` setting.